pkg/leader/leader_test.go

```diff
@@ -119,16 +119,23 @@ func Test_computeConfig(t *testing.T) {
 				cbs: leaderelection.LeaderCallbacks{},
 			},
 			envs: []env{
-				{key: "CATTLE_ELECTION_LEASE_DURATION", value: "1s"},
-				{key: "CATTLE_ELECTION_RENEW_DEADLINE", value: "2s"},
-				{key: "CATTLE_ELECTION_RETRY_PERIOD", value: "bomb"},
+				{key: leaseDurationEnvKey, value: "1s"},
+				{key: renewDeadlineEnvKey, value: "2s"},
+				{key: retryPeriodEnvKey, value: "bomb"},
 			},
 			want:    nil,
 			wantErr: true,
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
+			for _, e := range os.Environ() {
```

Comment:
First:
`os.Environ()` returns a list of envs in the form `key=value`. Instead I think you would want `os.Clearenv()`

Second
Unsetting all envs may be too aggressive, we should either explicitly unset the envs we care about Or capture the state of all the envs before our test and reset them after every test. 
Both of these options would avoid removing some standard shell envs like `PWD` that other unit tests/packages may be using.

Reply:
Right, addressed.